src/mscorlib/src/System/Threading/ThreadPool.cs

```diff
@@ -1195,27 +1193,17 @@ void MarkExecuted(bool aborted)
 #endif
 
         [SecurityCritical]
-        internal QueueUserWorkItemCallback(WaitCallback waitCallback, Object stateObj, bool compressStack, ref StackCrawlMark stackMark)
+        internal QueueUserWorkItemCallback(WaitCallback waitCallback)
         {
             callback = waitCallback;
-            state = stateObj;
-            if (compressStack && !ExecutionContext.IsFlowSuppressed())
-            {
-                // clone the exection context
-                context = ExecutionContext.Capture(
-                    ref stackMark,
-                    ExecutionContext.CaptureOptions.IgnoreSyncCtx | ExecutionContext.CaptureOptions.OptimizeDefaultCase);
-            }
         }
 
-        //
-        // internal test hook - used by tests to exercise work-stealing, etc.
-        //
-        internal QueueUserWorkItemCallback(WaitCallback waitCallback, Object stateObj, ExecutionContext ec)
+        [SecurityCritical]
+        protected virtual void ExecuteWorkItem()
```

Comment:
This is now introducing a virtual method call where there wasn't one before.  Previously processing each work item was a single interface call... now it's that interface call plus a virtual call.  We should avoid that.